### DISCUSSION TITLE
All pages being loaded at once

### DISCUSSION THREAD

**Discussion by chaseconey** [vuepress-theme-hope/vuepress-theme-hope#2474] [Q&A] [answered]
As per #2472, starting a discussion here.

**Issue:** When building and deploying the statically generated site, it seems that all 1000 or so pages are loaded at once at various times when navigating through the site. Can't seem to figure out the root cause of this; any help would be great!

## Additional context
It looks to be caused by the index page having a prefetch for every single other page.

![](https://user-images.githubusercontent.com/1449463/205354992-7bba223b-b809-4819-a9ad-c14f28f29b2e.png)

**Comment by chaseconey** [comment, marked as answer]
PWA was mentioned as a possibility in the issue, but I don't have the item in my config and it appears the default is `false`. Am I wrong about that?

**Reply by Mister-Hope** [reply]
You can set shouldPrefetch to `() => false`

**Reply by chaseconey** [reply]
Looks like that is defaulted to `false`? Would the option `shouldPreload` change this behavior? That seems to be defaulted to `true`.

**Reply by chaseconey** [reply]
Ok, can confirm that setting `shouldPrefetch` to `false` worked great. In the config types it says that the default is in fact `false`, is that just a typo?

**Reply by Mister-Hope** [reply]
Yes, we might forget to correct jsdocs

**Reply by Mister-Hope** [reply]
See https://github.com/vuepress/vuepress-next/pull/1191
